crates/libs/core/src/client/query_client.rs

```diff
@@ -37,48 +37,29 @@ impl QueryClient {
         queryDescription: &FABRIC_NODE_QUERY_DESCRIPTION,
         timeoutMilliseconds: u32,
     ) -> crate::sync::FabricReceiver<::windows_core::Result<IFabricGetNodeListResult2>> {
-        let (tx, rx) = crate::sync::oneshot_channel();
-        let com_cp = self.com.clone();
-        let callback = crate::sync::AwaitableCallback2::i_new(move |ctx| {
-            // Note we use the v2 api
-            let res = unsafe { com_cp.EndGetNodeList2(ctx) };
-            tx.send(res);
-        });
-        let ctx = unsafe {
-            self.com
-                .BeginGetNodeList(queryDescription, timeoutMilliseconds, &callback)
-        };
-        if ctx.is_err() {
-            let (tx2, rx2) = crate::sync::oneshot_channel();
-            tx2.send(Err(ctx.err().unwrap()));
-            rx2
-        } else {
-            rx
-        }
+        let com1 = &self.com;
+        let com2 = self.com.clone();
+        fabric_begin_end_proxy(
+            move |callback| unsafe {
+                com1.BeginGetNodeList(queryDescription, timeoutMilliseconds, callback)
+            },
+            move |ctx| unsafe { com2.EndGetNodeList2(ctx) },
```

Comment:
how will I do the custom error handling in this case?

Reply:
can I still get rx object here

Reply:
The semantic is unchanged as before.
The error of the begin and end com function will be propagated to the rx receiver in the content and returned to the user.
See the FabricReceiver content type is wrapped in "Result".